File: LeetCode/C++/200-299/225.cpp
#include <iostream>
#include <queue>
using namespace std;

class MyStack {
    queue<int> myQueue;
public:
    /** Initialize your data structure here. */
    MyStack() {

    }

    /** Push element x onto stack. */
    void push(int x) {
        myQueue.push(x);
        int size = myQueue.size();
        for(int i = 0 ; i < size -1;++i)
        {
            myQueue.push(pop());
        }
    }

    /** Removes the element on top of the stack and returns that element. */
    int pop() {
        int temp = top();
        myQueue.pop();
        return temp;
    }

    /** Get the top element. */
    int top() {
        return myQueue.front();
    }

    /** Returns whether the stack is empty. */
    bool empty() {
        return myQueue.empty();
    }

};


int main()
{
    MyStack myStack;
    myStack.push(1);
    myStack.push(2);
    myStack.push(3);
    cout << myStack.top() << endl;
    cout << myStack.pop() << endl;
    cout << myStack.top() << endl;
    cout << myStack.empty() << endl;
    myStack.pop();
    myStack.pop();
    cout << myStack.empty() << endl;
    return 0;
}

File: LeetCode/C#/200-299/225.cs
using System;
using System.Collections.Generic;
namespace LeetCode225
{
    public class MyStack {

        Queue<int> myQueue = new Queue<int>();
        /** Initialize your data structure here. */
        public MyStack() {
            
        }
        
        /** Push element x onto stack. */
        public void Push(int x) {
            myQueue.Enqueue(x);
            int size = myQueue.Count;
            for(int i = 0 ; i < size -1;++i)
            {
                myQueue.Enqueue(Pop());
            }
        }
        
        /** Removes the element on top of the stack and returns that element. */
        public int Pop() {
            return myQueue.Dequeue();
        }
        
        /** Get the top element. */
        public int Top() {
            return myQueue.Peek();
        }
        
        /** Returns whether the stack is empty. */
        public bool Empty() {
            return myQueue.Count == 0;
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            MyStack myStack = new MyStack();
            myStack.Push(1);
            myStack.Push(2);
            myStack.Push(3);
            Console.WriteLine(myStack.Top());
            Console.WriteLine(myStack.Pop());
            Console.WriteLine(myStack.Top());
            Console.WriteLine(myStack.Empty());
            myStack.Pop();
            myStack.Pop();
            Console.WriteLine(myStack.Empty());
        }
    }
}
